From 8c982aa72db6ce221f3c194fcde99b0aa1541653 Mon Sep 17 00:00:00 2001
From: Arpit Agarwal <arp@apache.org>
Date: Mon, 20 Mar 2017 11:29:23 -0700
Subject: [PATCH 2410/2748] HADOOP-14059. typo in s3a rename(self, subdir)
 error message. Contributed by Steve Loughran.

(cherry picked from commit 90e3283a0a0b9e6a5f82ea3e009b3c69ed8e8569)

Change-Id: I47bb762a38238bdc878d74d74c58698baee3ebef
---
 .../org/apache/hadoop/fs/s3a/S3AFileSystem.java    |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
index e78d4d6..56a7f4a 100644
--- a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
+++ b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
@@ -872,7 +872,7 @@ private boolean innerRename(Path source, Path dest)
       //Verify dest is not a child of the source directory
       if (dstKey.startsWith(srcKey)) {
         throw new RenameFailedException(srcKey, dstKey,
-            "cannot rename a directory to a subdirectory o fitself ");
+            "cannot rename a directory to a subdirectory of itself ");
       }
 
       List<DeleteObjectsRequest.KeyVersion> keysToDelete = new ArrayList<>();
-- 
1.7.9.5

